File: 2091-removing-minimum-and-maximum-from-array/2091-removing-minimum-and-maximum-from-array.cpp
class Solution {
public:
    int minimumDeletions(vector<int>& nums) {
        int n = nums.size();
        int mn = 0, mx = 0;

        for (int i = 1; i < n; i++) {
            if (nums[i] < nums[mn]) mn = i;
            if (nums[i] > nums[mx]) mx = i;
        }

        int left = min(mn, mx);
        int right = max(mn, mx);

        return min({right + 1, n - left, left + 1 + n - right});
    }
};

File: 2091-removing-minimum-and-maximum-from-array/2091-removing-minimum-and-maximum-from-array.py
class Solution:
    def minimumDeletions(self, nums):
        n = len(nums)
        mn = nums.index(min(nums))
        mx = nums.index(max(nums))

        a = min(mn, mx)
        b = max(mn, mx)

        return min(b + 1, n - a, a + 1 + n - b)